internal/services/appservice/helpers/function_app_schema.go

```diff
@@ -23,6 +24,152 @@ const (
+						"flex_function_app_deployment.0.storage_user_assigned_identity_access_enabled",
+						"flex_function_app_deployment.0.storage_connection_string_access_enabled",
+					},
+				},
+
+				"storage_access_key": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+					RequiredWith: []string{
+						"flex_function_app_deployment.0.storage_connection_string_access_enabled",
+					},
+				},
+
+				"storage_system_assigned_identity_access_enabled": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					ConflictsWith: []string{
+						"flex_function_app_deployment.0.storage_connection_string_access_enabled",
+						"flex_function_app_deployment.0.storage_user_assigned_identity_access_enabled",
+					},
+					ExactlyOneOf: []string{
+						"flex_function_app_deployment.0.storage_system_assigned_identity_access_enabled",
+						"flex_function_app_deployment.0.storage_user_assigned_identity_access_enabled",
+						"flex_function_app_deployment.0.storage_connection_string_access_enabled",
+					},
+				},
+
+				"storage_user_assigned_identity_access_enabled": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					RequiredWith: []string{
+						"flex_function_app_deployment.0.storage_user_assigned_identity_id",
+					},
+					ConflictsWith: []string{
+						"flex_function_app_deployment.0.storage_connection_string_access_enabled",
+						"flex_function_app_deployment.0.storage_system_assigned_identity_access_enabled",
+					},
+					ExactlyOneOf: []string{
+						"flex_function_app_deployment.0.storage_system_assigned_identity_access_enabled",
+						"flex_function_app_deployment.0.storage_user_assigned_identity_access_enabled",
+						"flex_function_app_deployment.0.storage_connection_string_access_enabled",
+					},
+				},
+
+				"storage_user_assigned_identity_id": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: commonids.ValidateUserAssignedIdentityID,
+					RequiredWith: []string{
+						"flex_function_app_deployment.0.storage_user_assigned_identity_access_enabled",
+					},
+				},
+
+				"runtime_name": {
+					Type:     pluginsdk.TypeString,
+					Required: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						string(webapps.RuntimeNameDotnetNegativeisolated),
+						string(webapps.RuntimeNameJava),
+						string(webapps.RuntimeNameNode),
+						string(webapps.RuntimeNamePowershell),
+						string(webapps.RuntimeNamePython),
+						string(webapps.RuntimeNameCustom),
+					}, false),
+				},
+
+				"runtime_version": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"instance_memory_in_mb": {
+					Type:         pluginsdk.TypeInt,
+					Optional:     true,
+					Default:      2048,
+					ValidateFunc: validation.IntInSlice([]int{512, 2048, 4096}),
```

Comment:
From our perspective, we would rather have new sizes available immediately and a control plane exception kick up, rather than having to wait for the AzureRM team to fix it each time a new size is available. My vote is for option 1 (no validation).

Reply:
removed the restriction.